pages/docs/reference/experimental.md

```diff
@@ -0,0 +1,232 @@
+```kotlin
+// library code
+@Experimental
+@Retention(AnnotationRetention.BINARY)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
+annotation class ExperimentalDateTime            // Experimental API marker
+
+@ExperimentalDateTime                            
+class DateProvider                              // Experimental class
+```
+</div>
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+```kotlin
+//client code
+@UseExperimental(ExperimentalDateTime::class)
+fun getDate(): Date {              // uses DateProvider; doesn't expose the experimental status
+    val dateProvider: DateProvider
+    // ...
+}
+
+fun displayDate() {
+    println(getDate())                     // OK: getDate() is not experimental
+}
+```
+</div>
+
+When somebody calls the function `getDate()`, they won't be informed about the experimental API used in its body. 
+
+To use an experimental API in all functions and classes in a file, add the file-level annotation `@file:UseExperimental` to the top of the file before the package specification and imports.
+ <div class="sample" markdown="1" theme="idea" data-highlight-only>
+ ```kotlin
+ //client code
+ @file:UseExperimental(ExperimentalDateTime::class)
+ ```
+ </div>
+
+### Module-wide use
+
+If you don't want to annotate every usage of experimental APIs in your code, you can accept the experimental status for your whole module. Module-wide use of experimental APIs can be propagating and non-propagating as well:
+* To accept the experimental status without propagation, compile the module with the argument `Xuse-experimental`, specifying the fully qualified name of the experimental API marker you use: `-Xuse-experimental=org.mylibrary.ExperimentalMarker`. Compiling with this argument has the same effect as if every declaration in the module having the annotation`@UseExperimental(ExperimentalMarker::class)`.
+* To accept and propagate the experimental status to your whole module, compile the module with the argument `-Xexperimental=org.mylibrary.ExperimentalMarker`. In this case, _every declaration_ in the module becomes experimental. The use of the module requires the acceptance of its experimental status as well.
+
+If you build your module with Gradle, you can add arguments like this:
+
+<div class="multi-language-sample" data-lang="groovy">
+<div class="sample" markdown="1" mode="groovy" theme="idea" data-lang="groovy">
+
+```groovy
+compileKotlin {
+    kotlinOptions {
+        freeCompilerArgs += "-Xuse-experimental=org.mylibrary.ExperimentalMarker"
+    }
+}
+```
+
+</div>
+</div>
+
+<div class="multi-language-sample" data-lang="kotlin">
+<div class="sample" markdown="1" mode="kotlin" theme="idea" data-lang="kotlin" data-highlight-only>
+
+```kotlin
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+// ...
+
+val compileKotlin: KotlinCompile by tasks
+
+compileKotlin.kotlinOptions.freeCompilerArgs += "-Xuse-experimental=org.mylibrary.ExperimentalMarker"

```

Comment:
I'm not sure that this is an idiomatic way to specify compiler arguments in Gradle Kotlin script. 
@h0tk3y Do you know a better way?

Reply:
Well, this is more or less the idiomatic way. An alternative that might looks arguably better as it doesn't require declaring the variable is:
```
tasks.getByName("compileKotlin") {
    kotlinOptions { /* ... */ }
}
```
We'll provide a way to access `kotlinOptions` through the *target – compilation* model soon.

Reply:
Updated the example slightly differently (also approved by @h0tk3y :)